### PR TITLE
Myriad/batch source subscription

### PR DESCRIPTION
**Pull request by nael-binary**

### CODE REVIEW

File: lib/Myriad/Transport/Redis.pm [hunk 1]
```diff
@@ -444,18 +441,27 @@ async method redis_from_pool {
...
+        $log->warnf('Max pool count reached! removing instance from pool, count now %d', $pool_count);
+        $self->remove_child($instance);
+    }
 }
 
 async method xreadgroup (@args) {
     my $instance = await $self->redis_from_pool;
-    # This should also be possible with a try/finally combination...
-    # but that's currently failing with the $redis_pool slot not being
-    # defined
```
Comment (nael-binary) [inline lib/Myriad/Transport/Redis.pm:455]: what do we mean by this? I don't think its still applicable

File: lib/Myriad/Transport/Redis.pm [hunk 2]
```diff
@@ -32,6 +32,7 @@ has $redis;
 has $redis_pool = [ ];
 has $wait_time = 15_000;
 has $batch_count = 50;
+has $max_pool_count = 10;
```
Comment (nael-binary) [inline lib/Myriad/Transport/Redis.pm:35]: Im not sure if this value is good, or how we should be able to control this value

Comment (eyadof) [reply lib/Myriad/Transport/Redis.pm:35]: Can you add it to `configure`? I think that would make it configurable by default.

File: lib/Myriad/Service/Implementation.pm [hunk 3]
```diff
@@ -219,16 +219,16 @@ async method start {
                     $log->tracef('Found emitter %s as %s', $method, $emitters->{$method});
                     my $spec = $emitters->{$method};
                     my $chan = $spec->{args}{channel} // die 'expected a channel, but there was none to be found';
-                    my $sink = $ryu->sink(
```
Comment (eyadof) [inline lib/Myriad/Service/Implementation.pm:222]: I think the concept here is `Emitter` is the opposite of `Receiver`

**Review by eyadof — APPROVED** [review]

File: lib/Myriad/Transport/Redis.pm (hunk 1 above)
Comment (tom-binary) [reply lib/Myriad/Transport/Redis.pm:455]: it means that we shouldn't need the `->on_ready` sub, and instead would be able to write:

```
my $instance = await $self->redis_from_pool;
try {
 ...
} finally {
 $self->return_redis_to_pool($instance);
}
```

except we can't due to limitations in try/finally, Object::Pad and/or async/await.

Comment (nael-binary) [reply lib/Myriad/Transport/Redis.pm:455]: oh I see, yep makes sense now. I will put back the comment since its still valid.

**Pull request merged**